### PR TITLE
V4100

### PR DESCRIPTION
**Pull request by Obaied**
- Compiled with AIR SDK 17
- Sample app should be compiled with AIR SDK 22 (anything >= 17)
- Add event tracking success/failure delegates
- Add session tracking success/failure delegates
- Support for standard deeplinking
- Support for sessions parameters (callback & partners)
- Update submodules (Android & iOS) to 4.10.0
- Write a wiki: https://github.com/adjust/sdk_issues/wiki/Adobe-AIR-SDK

### CODE REVIEW

File: src/com/adjust/sdk/Adjust.as [hunk 1]
```diff
@@ -2,222 +2,460 @@
...
-                adjustConfig.getLogLevel(), adjustConfig.getEventBufferingEnabled(),
-                adjustConfig.getAttributionCallbackDelegate() != null, adjustConfig.getDefaultTracker(),
-                sdkPrefix);
-
-        // For now, call onResume after onCreate.
-        extensionContext.call("onResume");
+    public static function start(adjustConfig:AdjustConfig):void {
+      if (extensionContext) {
+        trace("adjust warning: SDK already started");
+        //TODO: REMOVE RETURN
```
Comment (nonelse) [inline src/com/adjust/sdk/Adjust.as]: is it to remove or leave?

File: src/com/adjust/sdk/Adjust.as [hunk 2]
```diff
@@ -2,222 +2,460 @@
...
+      if (!extensionContext) {
+        trace(errorMessage);
+        return;
+      }
 
-        extensionContext.call("setReferrer", referrer);
+      extensionContext.call("setReferrer", referrer);
     }
 
-    public static function setDeviceToken(deviceToken:String):void {
```
Comment (nonelse) [inline src/com/adjust/sdk/Adjust.as]: Why was it removed? It's not a major release

**Comment by uerceg** [comment]
@adjust/integration

File: sample/Main.as [hunk 3]
```diff
@@ -0,0 +1,202 @@
...
+      buildButton(-1, "Track Partner Event", TrackPartnerClick);
+      buildButton(1, "Enable Offline Mode", EnableOfflineModeClick);
+      buildButton(2, "Disable Offline Mode", DisableOfflineModeClick);
+      buildButton(3, "Enable SDK", SetEnableClick);
+      buildButton(4, "Disable SDK", SetDisableClick);
+
+      IsEnabledTextField = buildButton(5, "Is SDK Enabled?", IsEnabledClick);
+
+      //--------Adjust Configuration-----------
+      var adjustConfig:AdjustConfig = new AdjustConfig("rb4g27fje5ej", Environment.SANDBOX);
```
Comment (Obaied) [inline sample/Main.as]: Fixed

File: src/com/adjust/sdk/Adjust.as (hunk 1 above)
Comment (Obaied) [inline src/com/adjust/sdk/Adjust.as]: I removed the todo.

File: src/com/adjust/sdk/Adjust.as (hunk 2 above)
Comment (Obaied) [inline src/com/adjust/sdk/Adjust.as]: Fixed.

**Pull request merged**